src/aiai_eval/task.py

```diff
@@ -54,7 +82,567 @@ def evaluate(self, model_id: str) -> Dict[str, dict]:
+        else:
+            itr = range(num_iter)
+
+        # Load the data collator
+        data_collator = self._load_data_collator(tokenizer)
+
+        scores = []
+        for idx in itr:
+            while True:
+                test_itr_scores = self._evaluate_pytorch_jax_single_iteration(
+                    idx=idx,
+                    model_config=model_config,
+                    tests=tests,
+                    data_collator=data_collator,
+                )
+                # If the iteration was successful then break the while-loop
+                if isinstance(test_itr_scores, dict):
+                    break
+
+                # Otherwise we encountered an error
+                else:
+                    raise InvalidEvaluation(str(test_itr_scores))
+
+            scores.append(test_itr_scores)
+
+        # Log scores
+        all_scores = log_scores(
+            dataset_name=self.dataset_task.pretty_dataset_name,
+            metric_configs=self.dataset_task.metrics,
+            scores=scores,
+            model_id=model_config.model_id,
+        )
+        return all_scores
+
+    def _evaluate_pytorch_jax_single_iteration(
+        self,
+        idx: int,
+        model_config: ModelConfig,
+        tests: Sequence[Dataset],
+        data_collator: DataCollator,
+    ) -> Union[dict, Exception]:
+        """Run a single iteration of a PyTorch/JAX benchmark.
+
+        Args:
+            idx (int):
+                The index of the current iteration.
+            model_dict (dict):
+                The model dictionary, with keys "model" and "tokenizer".
+            tests (list):
+                A list of bootstraped test datasets.
+            data_collator (DataCollator):
+                The data collator.
+
+        Returns:
+            dict or Exception:
+                The keys in the dict correspond to the metrics and values
+                the corresponding values.
+        """
+        scores = []
+        try:
+            # Set random seeds to enforce reproducibility of the randomly
+            # initialised weights
+            random.seed(703 + idx)
+            np.random.seed(703 + idx)
+            torch.manual_seed(703 + idx)
+            torch.cuda.manual_seed_all(703 + idx)
+
+            # Reinitialise a new model
+            model_dict = self._load_model(model_config=model_config)
+            model = model_dict["model"]
+
+            # Initialise compute_metrics function
+            compute_metrics = partial(
+                self._compute_metrics, id2label=model.config.id2label
+            )
+
+            # Get iteration data
+            test = tests[idx]
+
+            # Create dataloader
+            dataloader = DataLoader(
+                test, batch_size=32, shuffle=True, collate_fn=data_collator  # type: ignore
+            )
+
+            # Get model predictions
+            for batch in dataloader:
+                model_predictions = model(**batch).logits
```

Comment:
Doesn't work if `model` is not a Hugging Face model.

Reply:
Ah yes should have caught that...